cylc/flow/cycling/iso8601.py

```diff
@@ -647,111 +666,147 @@ def ingest_time(value, my_now=None):
         # Expanded year
         return value
 
+    value = value.replace(" ", "")  # remove extraneous whitespace
     parser = SuiteSpecifics.point_parser
     offset = None
 
-    if my_now is None:
-        my_now = parser.parse(get_current_time_string())
+    # work out what transformations we need to apply
+    is_prev_next = "next" in value or "previous" in value
+    is_offset = value.startswith("P") or value.startswith("-P")
+    is_truncated = False
+    if not (is_prev_next or is_offset):
+        timepoint = parser.parse(value)
+        is_truncated = timepoint.truncated
+
+    if is_prev_next or is_offset or is_truncated:
+        # we need information about the current time to process this
+        # this should only be attempted if the calendar is in
+        # gregorian mode
+        if CALENDAR.mode != Calendar.MODE_GREGORIAN:
+            raise CylcConfigError(
+                f'The value "{value}" can not be calculated '
+                f'in {CALENDAR.mode}'
+            )
+
+        if now is None:
+            now = parser.parse(get_current_time_string())
```

Comment:
Following the reference for `get_current_time_string()` to `wallclock.py` - is there a reason we use the `datetime` library instead of isodatetime to get the current time? Wouldn't the problem be solved by using isodatetime with the `CALENDAR` mode?

Reply:
An interesting idea, however, I'm not entirely sure it's actually possible to get the current date in the 360 day calendar, there being no real mapping between the two.

Reply:
So there is no defined "now" for the 360 day calendar (or it is only a relative calendar)? So when `metomi.isodatetime.data.get_timepoint_for_now()` returns `2021-04-28` that's completely arbitrary?

Reply:
Not completely arbitrary but it is essentially counting days from the start of the year when the calendar resets. Since one day in the 360 day calendar lasts longer than one day in the gregorian calendar it doesn't really make sense to say that today is such-and-such a date in the 360 day calendar. The 360 day calendar is essentially the calendar of a fictional planet.